extra_data/stacking.py

```diff
@@ -208,3 +208,22 @@ def asarray(self):
         for modno, data in self._data.items():
             arr[modno] = data
         return np.moveaxis(arr, 0, self._stack_axis)
+
+    def squeeze(self, axis=None):
+        """Drop axes of length 1 - see numpy.squeeze()"""
+        if axis is None:
+            slices = [0 if d == 1 else slice(None, None) for d in self.shape]
+        elif isinstance(axis, (int, tuple)):
+            if isinstance(axis, int):
+                axis = (axis,)
+
+            slices = [slice(None, None)] * self.ndim
+
+            for ax in axis:
+                if self.shape[ax] != 1:
```

Comment:
Great! LGTM except a nitpick, for out-of-bound axis this code raise `IndexError` while `np.squeeze` raises `AxisError`.

Reply:
True, should be fixed now.

Reply:
Thank you!